agent/engine/dependencygraph/graph.go

```diff
@@ -25,6 +25,15 @@ import (
 	"strings"
 )
 
+const (
+	StartCondition    = "START"
```

Comment:
lint will tell you that you need to add comments for all these globally visible consts. Also, on that note, do they need to be globally visible? Adding comments would still be a good idea, with a brief description of what these consts are used for and why they have these values.

Reply:
I may need global variables for the dependency conditions so that I can use them while writing functional tests. I have added comments to describe them. Thanks for the suggestion.

Reply:
marking these as not-global is probably the best bet here.

Reply:
If the functional tests take a dependency on internal agent states, we're probably doing something wrong wrt how we're writing these tests. I'd prefer them to local scoped for now. You can always expand the scope when/if needed.

Reply:
I haved changed it to local.